user/what-is-a-workspace.rst

```diff
@@ -0,0 +1,314 @@
+
+
+Put this content into ``src/foo/setup.py``:
+
+.. code-block:: python
+
+	from setuptools import setup
+
+	setup()
+
+Put this content into ``src/foo/setup.cfg``:
+
+.. code-block:: ini
+
+	[metadata]
+	name = foo
+	version = 0.1.0
+
+	[options]
+	py_modules =
+	  foo
+	zip_safe = true
+
+	[options.extras_require]
+	test =
+	  pytest
+
+	[tool:pytest]
+	junit_suite_name = foo
+
+Put this content into ``src/foo/foo.py``:
+
+.. code-block:: python
+
+	def foo_func():
+		print('Hello from foo.py')
+		return True
+
+
+Build artifacts
+***************
+
+The software build process often produces intermediate build artifacts.
+They are usually not used directly, but keeping them around makes subsequent builds faster.
+Colcon always directs packages to build out-of-source, meaning the build artifacts are put into a directory separate from the source code.
+Every package gets its own build directory, but all build directories are put into a single base directory.
+That single directory is called the ``build base``, and by default it's named  ``build`` at the root of the workspace.
+
+.. note::
+
+	You can change where build artifacts are put using the ``--build-base`` option to ``colcon build``.
+
+Lets build the software and see its build artifacts.
+
+.. code-block:: bash
+
+	# Make sure you run this command from the root of the workspace!
+	colcon build
+
+You'll see these new directories: ``build``, ``install``, and ``log``.
+
+::
+
+	ws
+	├── build
+	│	├── COLCON_IGNORE
+	│	└── foo/...
+	├── install/...
+	├── log/...
+	└── src
+	    └── foo
+	        ├── foo.py
+	        ├── setup.cfg
+	        └── setup.py
+
+Notice the ``build`` directory has a subdirectory ``foo`` and a file ``COLCON_IGNORE``.
+The ``foo`` subdirectory has all the build artifacts produced when building ``foo``.
+The ``COLCON_IGNORE`` file tells colcon there are no software packages in this directory.
+
+Logs
+****
+
+If you've built software before you know there can be a lot of console output, but you might have noticed not much was output when you ran ``colcon build``.
+This output was instead written to the ``log`` directory.
+
+.. note::
+
+	The ``--event-handlers`` argument can be used to output build logs to the console. For example, ``colcon build --event-handlers console_direct+`` will output everything in real time.
+
+
+Let's look at the ``log`` directory.
+
+::
+
+	log
+	├── build_2022-05-20_11-50-03
+	│	├── events.log
+	│	├── foo
+	│	│	├── command.log
+	│	│	├── stderr.log
+	│	│	├── stdout.log
+	│	│	├── stdout_stderr.log
+	│	│	└── streams.log
+	│	└── logger_all.log
+	├── COLCON_IGNORE
+	├── latest -> latest_build
+	└── latest_build -> build_2022-05-20_11-50-03
+
+
+The directory ``log/build_<date and time>`` contains all logs from the invocation of ``colcon build``.
+A new directory is created every time ``colcon build`` is run.
+The symlink ``latest_build`` always point to the most recent build.
+
+..
+	TODO(sloretz) what is events.log and logger_all.log?
```

Comment:
As mentioned about, this is described in https://github.com/colcon/colcon.readthedocs.org/blob/main/user/how-to.rst#log-files-of-past-invocations , so we should move it from there.